core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala

```diff
@@ -78,6 +82,26 @@ class AppStatusStoreSuite extends SparkFunSuite {
     assert(store.count(classOf[CachedQuantile]) === 2)
   }
 
+  test("InMemoryStore should build index from Stage ID to Task data") {
+    val store = new InMemoryStore()
+    (0 until 1000).map { sId =>
+       (0 until 1000).map { taskId =>
+         val task = newTaskData(sId * 1000 + taskId, "SUCCESS", sId)
+         store.write(task)
+       }
+    }
+    val appStatusStore = new AppStatusStore(store)
+    failAfter(Span(200, Millis)) {
+      appStatusStore.taskSummary(1, attemptId, Array(0, 0.25, 0.5, 0.75, 1))
+    }
+    val stageIds = Seq(1, 11, 66, 88)
+    val stageKeys = stageIds.map(Array(_, attemptId))
+    failAfter(Span(10, Millis)) {
```

Comment:
up to my experience, this can be flaky as it assumes some code should be executed within some time. I think the benchmark result in PR description is good enough.